.github/workflows/security.yml

```diff
@@ -30,26 +30,52 @@ jobs:
   black-duck-scans:
     name: Run Black Duck Scans
     runs-on: [self-hosted, Linux]
+    strategy:
+      fail-fast: false
+      matrix:
+        PACKAGE: ["code-editor", "core", "icons", "lab", "shared", "styles", "uno-preset", "viz"]
 
     steps:
       - name: Checkout
         uses: actions/checkout@v4
 
-      - name: Set up Java
-        uses: actions/setup-java@v4
+      - name: Install Node.js
+        uses: actions/setup-node@v4
         with:
-          java-version: "11"
-          distribution: "adopt"
+          node-version: 18
 
-      - name: Install Dependencies
-        uses: ./.github/actions/install-dependencies
+      # ==========================
+      # code-editor (npm package)
+      # ==========================
 
-      - name: Build
-        run: npm run build
+      # Install dependencies inside each package so blackduck can scan them
+      # To do this we need to remove the package.json and package-lock.json from the root
+      - name: Prepare packages for Blackduck scan
+        uses: lumada-common-services/gh-composite-actions@1.9.0
+        with:
+          command: |
+            rm -rf node_modules package.json package-lock.json && \
+            cd packages/${{ matrix.PACKAGE }} && npm i
+
+      - name: Get BlackDuck project name
+        run: echo "BLACKDUCK_PROJECT=$(./.github/scripts/getBlackduckProject.sh ${{ matrix.PACKAGE }})" >> $GITHUB_ENV
+
+      - name: Get BlackDuck project version
+        run: echo "PROJECT_VERSION=$( npm pkg get version --workspaces=false | tr -d '"' )" >> $GITHUB_ENV
+        working-directory: packages/${{ matrix.PACKAGE }}
+
+      - name: Load blackduck project properties
+        shell: bash
+        run: echo "BLACKDUCK_ARGS=$(awk '{print}' ORS=' ' .github/security-configurations/blackduck-${{ matrix.PACKAGE }}.properties)" >> $GITHUB_ENV
 
-      - name: Run Detect Scans
-        run: .github/scripts/blackduck-detect.sh
+      - name: Blackduck Scan
+        uses: lumada-common-services/gh-composite-actions@1.9.0
         env:
-          BLACKDUCK_URL: "${{ secrets.BLACKDUCK_URL }}"
-          BLACKDUCK_TOKEN: "${{ secrets.BLACKDUCK_TOKEN }}"
-          DETECT_VERSION: 8
+          BLACKDUCK_DOCKER_USERNAME: hvservices-service-cicd
```

Comment:
Should we create a secret for this value?

Reply:
I don't think it's worth it - it's just a generic CI username.

Reply:
I don't think it's worth it - it's just a generic CI username.